src/expr.rs

```diff
@@ -245,8 +245,8 @@ fn format_expr(expr: &ast::Expr,
         ast::ExprKind::InlineAsm(..) => {
             wrap_str(context.snippet(expr.span), context.config.max_width, shape)
         }
-        // FIXME(#1537)
-        ast::ExprKind::Catch(..) => unimplemented!(),
+        // 6 = `catch `
+        ast::ExprKind::Catch(ref block) => block.rewrite(&context, try_opt!(shape.sub_width(6))),
```

Comment:
I have few questions:
1. Should `do catch { ... }` stay in a single line as `unsafe` when it is short?
2. I found an example without `do` (`let x = match catch { ... }`). How should I treat these differently?